src/core/connected_clients.rs

```diff
@@ -11,16 +11,53 @@ use bevy::{
 use crate::core::{replicon_tick::RepliconTick, ClientId};
 use client_visibility::ClientVisibility;
 
-/// Stores information about connected clients.
+#[derive(Resource, Default)]
+pub struct ConnectedClients {
+    clients: Vec<ClientId>,
+    pub(crate) replicate_after_connect: bool,
+}
+
+impl ConnectedClients {
+    pub(crate) fn new(replicate_after_connect: bool) -> Self {
+        Self {
+            clients: Default::default(),
+            replicate_after_connect,
+        }
+    }
+
+    /// Gets an iterator over all [`ClientId`]s currently tracked.
+    pub fn iter_client_ids(&self) -> impl Iterator<Item = ClientId> + '_ {
```

Comment:
I don't exactly like this because this change makes it a breaking update for bevy_replicon_renet, which uses `ConnectedClients::iter_client_ids`. It's a small change but would just need an extra PR to get it updated, otherwise there are no other changes required to bevy_replicon_renet.

Reply:
It's okay, I will bump `bevy_replicon` version to 0.x, will require updating `bevy_replicon_renet` anyway. But don't worry, I will do it myself :)

Reply:
Alright, made it `iter`